cot/src/admin.rs

```diff
@@ -144,16 +173,57 @@ async fn view_model(request: Request) -> cot::Result<Response> {
         model: &'a dyn AdminModelManager,
         #[debug("..")]
         objects: Vec<Box<dyn AdminModel>>,
+        page: usize,
+        page_size: &'a usize,
+        total_object_counts: usize,
+        total_pages: usize,
     }
 
     let model_name: String = request.path_params().parse()?;
     let manager = get_manager(&request, &model_name)?;
 
+    let query_params: HashMap<String, String> = request
+        .uri()
+        .query()
+        .map(|q| {
+            query_pairs(&Bytes::copy_from_slice(q.as_bytes()))
+                .map(|(k, v)| (k.to_string(), v.to_string()))
+                .collect()
+        })
+        .unwrap_or_default();
+
+    let page: usize = query_params
+        .get("page")
+        .map_or(1, |p| p.parse().unwrap_or(1));
+
+    let limit = query_params
+        .get("page_size")
+        .map_or(10, |p| p.parse().unwrap_or(10));
+
+    let total_object_counts = manager.get_total_object_counts(&request).await?;
+    let total_pages = total_object_counts.div_ceil(limit);
+
+    if page == 0 || page > total_pages {
+        return Ok(Response::new_html(
+            StatusCode::NOT_FOUND,
+            Body::fixed(include_str!("../templates/404.html")),
+        ));
```

Comment:
This way a user-defined 404 page can be displayed (or a generic Cot page in case the debug mode is on).

```suggestion
        return Err(Error::not_found_message(format!("page {page} not found")));
```